mirascope/base/vectorstores.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Optional, Union
+
+from pydantic import BaseModel
+
+from mirascope.base.chunkers import BaseChunker
+from mirascope.base.embedders import BaseEmbedder
+from mirascope.openai.embedders import OpenAIEmbedder
+
+from .types import BaseVectorStoreParams, Document
+
+
+class BaseVectorStore(BaseModel, ABC):
+    vectorstore_api_key: Optional[str] = None
+    index_name: ClassVar[Optional[str]] = None
+    chunker: ClassVar[BaseChunker] = BaseChunker()
+    embedder: ClassVar[Optional[BaseEmbedder]] = OpenAIEmbedder()
+    vectorstore_params: ClassVar[BaseVectorStoreParams] = BaseVectorStoreParams()
+
+    @abstractmethod
+    def get_documents(
```

Comment:
do we prefer `get_documents` to `query` and `add_documents` to `upload`?

Reply:
its doing more than just `query`. upload is not really a database word so i'm against that specifically.

Reply:
Ok. I don't love `get_documents` because it loses the intent behind the word `query` and sounds more like getting all of the documents in the vector store.

What more beyond `query` is it doing?

As for not liking `upload` what about just `add`?

So it would be `store.query` and `store.add`